src/app/_services/task/task.service.ts

```diff
@@ -0,0 +1,57 @@
+import {Injectable} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+import {HttpClient, HttpErrorResponse} from '@angular/common/http';
+import {catchError, map} from 'rxjs/operators';
+import {Task} from '../../_models';
+import {ErrorObservable} from 'rxjs/observable/ErrorObservable';
+
+export enum TaskError {
+    ERR_OK = 200,
+    ERR_SERVER = 500,
+    ERR_NODE_N_SWARM = 503,
+    ERR_NO_TASK = 404,
+}
+
+@Injectable()
+export class TaskService {
+
+    constructor(private http: HttpClient) {
+    }
+
+    public getTasks(): Observable<Task[]> {
+        return this.http.get('http://localhost:8080/docker/tasks', {responseType: 'json'})
```

Comment:
Is it possible to store the address of the Docks API in a more abstract way?
The address will be given when the container is deployed using either a ENV or `docker config`.
The address could be for example `http://<ip>:<port>` or `https://docks.example.com:<port>`

Reply:
I haven't had a chance to look at this properly yet. I thought about creating another service to store a few configuration options so that it can be changed globally at a single point.

I think if the frontend is deployed with the backend using a compose file it should be able to refer to the api via a hostname which is then resolved to the correct container.  Would that be ideal?

Reply:
I know Docker Swarm has its own DNS, we might be able to use that to resolve the hostname of the Docks API server. See [overlay networks](https://docs.docker.com/network/overlay/)

I am happy to merge this PR (once reviewed by other members) even with the localhost URL.
As long as the endpoint URL is wrapped in a service as you did, it will be easy to change to more sustainable solution.

Reply:
I'll refactor any of the services that use the static address once I've decided on how to implement this.

I'll have a read and get back to you.

Reply:
I checked and the API is reachable as docks-api.

Would it be alright adding a service for the configuration and having it return the api hostname as docks-api?

Reply:
Yes that sounds good

Reply:
No wait, the browser has ti resolve that address. It cannot be a swarm overlay network. It will have to be the public IP and port of the web server

Reply:
Ah web development x)

This is just taking a note, I believe in class you said we'd just allow whoever was deploying the software to configure this, perhaps via a environmental variable?

Reply:
Yes and environmental variable would be easy to configure from the docker-compose file. If you can read the ENV and when it is not present use localhost that would be good

Reply:
Nevermind the above comment, again forgot that docks-ui is run completely in the browser.